### PR TITLE
handle app notification settings, and set new field on new message frontend notifications about OS notifications CORE-5601

### PR DESCRIPTION
**Pull request by mmaxim**
Patch does the following:

1.) Adds a new local endpoint for setting app notification settings.
2.) Handles new app notification changed Gregor message. Includes maintaining inbox cache correctly, and sending a message to the frontend that the settings have changed.
3.) Set a bit on new message frontend notifications informing it whether or not to show the desktop toast. cc @chrisnojima about this.
4.) Bunch of protocol changes.

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
The commits bcb5dd1fe154...9f3606e663db introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1499830771605-bcb5dd1fe154-9f3606e663db-linux/Markdown-emoji-bcb5dd1fe154.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1499830771605-bcb5dd1fe154-9f3606e663db-linux/Markdown-emoji-9f3606e663db.png)  
   Markdown-emoji comparison
</summary>

File: go/chat/push.go [hunk 1]
```diff
@@ -473,6 +511,26 @@ func (g *PushHandler) Activity(ctx context.Context, m gregor.OutOfBandMessage) (
 			if g.badger != nil && nm.UnreadUpdate != nil {
 				g.badger.PushChatUpdate(*nm.UnreadUpdate, nm.InboxVers)
 			}
+		case "setAppNotificationSettings":
```
Comment (patrickxb) [inline go/chat/push.go]: didn't you change these to constants on the other side?  any way to use those?

Comment (mmaxim) [inline go/chat/push.go]: Yeah I should just define them in client and vendor them into Gregor, I'll do that now. Should have just made this an enum from the get-go, not sure why I didn't, but too annoying to change now.

**Comment by keybase-ci-visdiff** [comment]
The commits bcb5dd1fe154...31d85f3d8f5d introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1499878075461-bcb5dd1fe154-31d85f3d8f5d-linux/Markdown-emoji-bcb5dd1fe154.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1499878075461-bcb5dd1fe154-31d85f3d8f5d-linux/Markdown-emoji-31d85f3d8f5d.png)  
   Markdown-emoji comparison
</summary>

**Pull request merged**